code_generator/GdkCodeGenerator/Templates/UnityReferenceTypeProviderContent.tt

```diff
@@ -0,0 +1,67 @@
+<#@ template language="C#" #>
+<#@ output extension=".cs" #>
+public class <#= Name #>Provider 
+{
+    private static readonly Dictionary<uint, <#= TypeName #>> Storage = new Dictionary<uint, <#= TypeName #>>();
+    private static readonly Dictionary<uint, global::Unity.Entities.World> WorldMapping = new Dictionary<uint, Unity.Entities.World>();
```

Comment:
Does it make more sense to swap this round - world to list of uint ? I think in all access cases the world is the bit of information you know.

Reply:
I think with the restrictions placed on removal elsewhere, that may be possible. Will TAL at optimising that

Reply:
Discussed offline: this might be premature optimisation for a rare use case. Will be easy to apply this optimisation in the future if needed. Will not make this change at this time